components/ingest-service/config/config_mgmt.go

```diff
@@ -226,85 +215,37 @@ func (manager *Manager) GetMissingNodesForDeletionSchedulerConfig() JobConfig {
 
 // UpdateDeleteNodesSchedulerConfig - update the delete node config
 func (manager *Manager) UpdateDeleteNodesSchedulerConfig(jConfig JobConfig) error {
-	errc := make(chan error)
-
-	updateFunc := func(config aggregateConfig) aggregateConfig {
+	return manager.updateConfig(func(config aggregateConfig) (aggregateConfig, error) {
 		config.JobsConfig[DeleteNodes] = jConfig
-		errc <- saveToFile(config)
-		return config
-	}
-
-	manager.send(updateFunc)
-	return <-errc
+		return config, nil
+	})
 }
 
 // UpdateNodesMissingSchedulerConfig update the missing nodes for deletion task
 func (manager *Manager) UpdateNodesMissingSchedulerConfig(jConfig JobConfig) error {
-	errc := make(chan error)
-
-	updateFunc := func(config aggregateConfig) aggregateConfig {
+	return manager.updateConfig(func(config aggregateConfig) (aggregateConfig, error) {
 		config.JobsConfig[NodesMissing] = jConfig
-		errc <- saveToFile(config)
-		return config
-	}
-
-	manager.send(updateFunc)
-	return <-errc
+		return config, nil
+	})
 }
 
 // UpdateMissingNodesForDeletionSchedulerConfig update the missing nodes for deletion config
 func (manager *Manager) UpdateMissingNodesForDeletionSchedulerConfig(jConfig JobConfig) error {
-	errc := make(chan error)
-
-	updateFunc := func(config aggregateConfig) aggregateConfig {
+	return manager.updateConfig(func(config aggregateConfig) (aggregateConfig, error) {
 		config.JobsConfig[MissingNodesForDeletion] = jConfig
-		errc <- saveToFile(config)
-		return config
-	}
-
-	manager.send(updateFunc)
-	return <-errc
+		return config, nil
+	})
 }
 
-func (manager *Manager) send(updateFunc func(aggregateConfig) aggregateConfig) {
-	manager.updateQueue <- updateFunc
+func (manager *Manager) updateConfig(updateFunc func(aggregateConfig) (aggregateConfig, error)) error {
+	return manager.baseConfigManager.UpdateConfig(func(config interface{}) (interface{}, error) {
+		return updateFunc(config.(aggregateConfig))
+	})
 }
 
-func saveToFile(config aggregateConfig) error {
```

Comment:
Moved the core logic to lib/config/config_mgmt.go for authz-service and compliance-service use.